core/services/relay/evm/write_target.go

```diff
@@ -69,7 +81,192 @@ func NewWriteTarget(ctx context.Context, relayer *Relayer, chain legacyevm.Chain
+	if err != nil {
+		return "", err
+	}
+
+	if reportMetadata.Version != 1 {
+		return "", fmt.Errorf("unsupported report version: %d", reportMetadata.Version)
+	}
+
+	if hex.EncodeToString(reportMetadata.WorkflowExecutionID[:]) != rawRequest.Metadata.WorkflowExecutionID {
+		return "", fmt.Errorf("WorkflowExecutionID in the report does not match WorkflowExecutionID in the request metadata. Report WorkflowExecutionID: %+v, request WorkflowExecutionID: %+v", hex.EncodeToString(reportMetadata.WorkflowExecutionID[:]), rawRequest.Metadata.WorkflowExecutionID)
+	}
+
+	// case-insensitive verification of the owner address (so that a check-summed address matches its non-checksummed version).
+	if !strings.EqualFold(hex.EncodeToString(reportMetadata.WorkflowOwner[:]), rawRequest.Metadata.WorkflowOwner) {
+		return "", fmt.Errorf("WorkflowOwner in the report does not match WorkflowOwner in the request metadata. Report WorkflowOwner: %+v, request WorkflowOwner: %+v", hex.EncodeToString(reportMetadata.WorkflowOwner[:]), rawRequest.Metadata.WorkflowOwner)
+	}
+
+	// workflowNames are padded to 10bytes
+	decodedName, err := hex.DecodeString(rawRequest.Metadata.WorkflowName)
+	if err != nil {
+		return "", err
+	}
+	var workflowName [10]byte
+	copy(workflowName[:], decodedName)
+	if !bytes.Equal(reportMetadata.WorkflowName[:], workflowName[:]) {
+		return "", fmt.Errorf("WorkflowName in the report does not match WorkflowName in the request metadata. Report WorkflowName: %+v, request WorkflowName: %+v", hex.EncodeToString(reportMetadata.WorkflowName[:]), hex.EncodeToString(workflowName[:]))
+	}
+
+	if hex.EncodeToString(reportMetadata.WorkflowCID[:]) != rawRequest.Metadata.WorkflowID {
+		return "", fmt.Errorf("WorkflowID in the report does not match WorkflowID in the request metadata. Report WorkflowID: %+v, request WorkflowID: %+v", reportMetadata.WorkflowCID, rawRequest.Metadata.WorkflowID)
+	}
+
+	if !bytes.Equal(reportMetadata.ReportID[:], r.Inputs.SignedReport.ID) {
+		return "", fmt.Errorf("ReportID in the report does not match ReportID in the inputs. reportMetadata.ReportID: %x, Inputs.SignedReport.ID: %x", reportMetadata.ReportID, r.Inputs.SignedReport.ID)
+	}
+
+	return r.Config.Address, nil
+}
+
+func decodeReportMetadata(data []byte) (metadata ReportV1Metadata, err error) {
+	if len(data) < metadata.Length() {
+		return metadata, fmt.Errorf("data too short: %d bytes", len(data))
+	}
+	return metadata, binary.Read(bytes.NewReader(data[:metadata.Length()]), binary.BigEndian, &metadata)
+}
+
+func (rm ReportV1Metadata) Encode() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	err := binary.Write(buf, binary.BigEndian, rm)
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
+
+func (rm ReportV1Metadata) Length() int {
+	bytes, err := rm.Encode()
+	if err != nil {
+		return 0
+	}
+	return len(bytes)
+}
+
+// Note: This should be a shared type that the OCR3 package validates as well
+type ReportV1Metadata struct {
```

Comment:
This should be the same type as here: https://github.com/smartcontractkit/chainlink-evm/pull/2/files#diff-a1c773745f91fd6fdd0d21e79154388d130223cc683aeff1e6bebe7fd1946520R14

How about we reuse it?

Reply:
I'm going to add this to the future tickets since this would require creating a new `chainlink-common` PR, which may add more friction.